### PR TITLE
Issue #124894 fix: TJNotification integration with jticketing-backend - Unable to sort all columns

### PR DESCRIPTION
**Pull request by samadhansalunkhe**
- [x] Have you followed the [Contribution Guidenlines](https://github.com/ekstep/Community-Portal-Joomla/wiki/Contribution-guidenlines)?
- [x] Have you done PHPCS?
- [ ] Have you done JSHINT?
- [ ] Have you checked for typos in language files (if any)?

-----

### CODE REVIEW

File: src/com_tjnotifications/admin/models/notifications.php [hunk 1]
```diff
@@ -31,8 +31,9 @@ public function __construct($config = array())
 				'id',
 				'client',
 				'key',
-				'state', 'a.state'
-
+				'state',
+				'a.state',
```
Comment (ankush-maherwal) [inline src/com_tjnotifications/admin/models/notifications.php]: seems a.state is not used

check and remove it

File: src/com_tjnotifications/admin/models/notifications.php [hunk 2]
```diff
@@ -65,36 +66,48 @@ protected function getListQuery()
...
-				{
-					$query->where($db->quoteName('client') . ' = ' . $db->quote($client));
-				}
-			}
+			$client = $this->getState('filter.client');
+			$key = $this->getState('filter.key');
 
-			// Fot getting templates
-			if (!empty($client) && !empty($key))
+			if (!empty($client) && empty($key))
```
Comment (ankush-maherwal) [inline src/com_tjnotifications/admin/models/notifications.php]: why we have removed ! condition?

File: src/com_tjnotifications/admin/models/notifications.php [hunk 3]
```diff
@@ -65,36 +66,48 @@ protected function getListQuery()
...
+			$query->where($db->quoteName('client') . ' = ' . $db->quote($client) . ' AND ' . $db->quoteName('key') . ' = ' . $db->quote($key));
+			$query->order('`key` ASC');
+		}
+
+		$orderCol  = $this->getState('list.ordering');
+		$orderDirn = $this->getState('list.direction');
+
+		if ($orderCol == 'key')
+		{
+			$query->order('`key`' . ' ' . $db->escape($orderDirn));
```
Comment (ankush-maherwal) [inline src/com_tjnotifications/admin/models/notifications.php]: use quoteName

File: src/com_tjnotifications/admin/models/notifications.php (hunk 1 above)
Comment (samadhansalunkhe) [reply src/com_tjnotifications/admin/models/notifications.php]: Removed it. Working fine.

File: src/com_tjnotifications/admin/models/notifications.php (hunk 3 above)
Comment (samadhansalunkhe) [reply src/com_tjnotifications/admin/models/notifications.php]: Done

File: src/com_tjnotifications/admin/models/notifications.php [hunk 4]
```diff
@@ -149,6 +154,26 @@ public function getTemplate($client,$key)
 	 */
 	protected function populateState($ordering = 'id', $direction = 'asc')
 	{
+		$app    = JFactory::getApplication();
+
+		$ordering = $app->input->get('filter_order', 'id', 'STRING');
+		$direction = $app->input->get('filter_order_Dir', 'desc', 'STRING');
+
+		if (!empty($ordering))
```
Comment (manojLondhe) [inline src/com_tjnotifications/admin/models/notifications.php]: @samadhansalunkhe 

Add 
```

if (!empty($ordering) && in_array($ordering, $this->filter_fields))
{
	$this->setState('list.ordering', $ordering);
}
```

Comment (samadhansalunkhe) [reply src/com_tjnotifications/admin/models/notifications.php]: Added

**Pull request merged**